packages/dma-contracts/tasks/verify-operations/index.ts

```diff
@@ -0,0 +1,232 @@
+
+import {
+  ActionDefinition,
+  ActionsDatabase,
+  ActionValidationResult,
+  getOperationRegistry,
+  getServiceRegistry,
+  isInvalidAddress,
+  OperationDefinition,
+  OperationDefinitionGetter,
+  VerificationResult,
+} from '../common'
+
+function validateActionHashes(
+  operationHashes: string[],
+  operationOptionals: boolean[],
+  actionDefinitions: ActionDefinition[],
+  actionsDatabase: ActionsDatabase,
+): ActionValidationResult {
+  let actionsValidated = true
+  let actionsErrorMessage: string | undefined = undefined
+
+  for (let actionIndex = 0; actionIndex < actionDefinitions.length; actionIndex++) {
+    if (
+      actionDefinitions[actionIndex].hash !== operationHashes[actionIndex] ||
+      actionDefinitions[actionIndex].optional !== operationOptionals[actionIndex]
+    ) {
+      actionsValidated = false
+
+      const actionDefinitionName = actionsDatabase.getActionName(
+        actionDefinitions[actionIndex].hash,
+      )
+      const operationHashName = actionsDatabase.getActionName(operationHashes[actionIndex])
+
+      actionsErrorMessage = `Action ${actionIndex} expected hash ${
+        actionDefinitionName ? actionDefinitionName : actionDefinitions[actionIndex].hash
+      } is different from registry ${
+        operationHashName ? operationHashName : operationHashes[actionIndex]
+      }`
+      break
+    }
+  }
+
+  return {
+    success: actionsValidated,
+    errorMessage: actionsErrorMessage,
+  }
+}
+
+async function validateDeployedActions(
+  actionsDatabase: ActionsDatabase,
+  serviceRegistry: ServiceRegistry,
+  actionDefinitions: ActionDefinition[],
+): Promise<ActionValidationResult> {
+  const actionsNotDeployed: string[] = []
+
+  for (let actionIndex = 0; actionIndex < actionDefinitions.length; actionIndex++) {
+    const actionAddress = await serviceRegistry.getServiceAddress(
+      actionDefinitions[actionIndex].hash,
+    )
+
+    if (isInvalidAddress(actionAddress)) {
+      const actionName = actionsDatabase.getActionName(actionDefinitions[actionIndex].hash)
+      actionsNotDeployed.push(actionName)
+      break
+    }
+  }
+
+  if (actionsNotDeployed.length > 0) {
+    return {
+      success: false,
+      errorMessage: `missing actions: [${actionsNotDeployed.join(', ')}]`,
+    }
+  } else {
+    return {
+      success: true,
+    }
+  }
+}
+
+async function validateOperations(
+  actionsDatabase: ActionsDatabase,
+  serviceRegistry: ServiceRegistry,
+  operationRegistry: OperationsRegistry,
+  operationDefinitions: OperationDefinition[],
+): Promise<VerificationResult> {
+  let totalValidated = 0
+  let totalEntries = 0
+
+  for (const operationDefinition of operationDefinitions) {
+    totalEntries++
+
+    let operationHashes: string[]
+    let operationOptionals: boolean[]
+    try {
+      ;[operationHashes, operationOptionals] = await operationRegistry.getOperation(
+        operationDefinition.name,
+      )
+    } catch (e: any) {
```

Comment:
Could we use `unknown` instead of `any`?

Reply:
Sure! What the idea behind it? Why is `unknown` better?

Reply:
Basically it's better for runtime errors. 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type
You can't access any properties without a type assertion.
You can also use syntax to check specific type definitions: 
```
function checkIfItHasMessage(x: unknown): x is { message: string } {
  if (typeof x !== 'object' || x === null) return false
  return 'message' in x
}
```